code/modules/projectiles/guns/lever_action.dm

```diff
@@ -327,6 +347,142 @@ their unique feature is that a direct hit will buff your damage and firerate
 						)
 	map_specific_decoration = TRUE
 	flags_gun_features = GUN_CAN_POINTBLANK|GUN_INTERNAL_MAG|GUN_AMMO_COUNTER
+	flags_gun_lever_action = MOVES_WHEN_LEVERING|DANGEROUS_TO_ONEHAND_LEVER
 
 /obj/item/weapon/gun/lever_action/r4t/set_gun_attachment_offsets()
 	attachable_offset = list("muzzle_x" = 33, "muzzle_y" = 19, "rail_x" = 11, "rail_y" = 21, "under_x" = 24, "under_y" = 16, "stock_x" = 15, "stock_y" = 14)
+
+//===================THE XM88===================\\
+
+#define FLOATING_PENETRATION_TIER_0	0
+#define FLOATING_PENETRATION_TIER_1	1
+#define FLOATING_PENETRATION_TIER_2	2
+#define FLOATING_PENETRATION_TIER_3	3
+#define FLOATING_PENETRATION_TIER_4	4
+
+/obj/item/weapon/gun/lever_action/xm88
+	name = "\improper XM88 heavy rifle"
+	desc = "An experimental man-portable anti-material rifle originally developed by ARMAT Battlefield Systems for the government of the state of Greater Brazil for use in the Favela Wars (2161 - Ongoing) against mechanized infantry. Chambered in .458 SOCOM, the platform features an onboard computerized targeting system, sensor array, and an electronic autoloader; these features work in tandem to reduce and render inert armor on the users target with successive hits. The Almayer was issued a small amount of XM88s while preparing for Operation Swamp Hopper with the USS Nan-Shan.\nIt has a special property - when you obtain multiple direct hits in a row, its armour penetration and damage will increase."
```

Comment:
```suggestion
	desc = "An experimental man-portable anti-material rifle originally developed by ARMAT Battlefield Systems for the government of the state of Greater Brazil for use in the Favela Wars (2161 - Ongoing) against mechanized infantry. Chambered in .458 SOCOM, the platform features an onboard computerized targeting system, sensor array, and an electronic autoloader; these features work in tandem to reduce and render inert armor on the users target with successive hits. The Almayer was issued a small amount of XM88s while preparing for Operation Swamp Hopper with the USS Nan-Shan. It must be manually chambered for every shot.\nIt has a special property - when you obtain multiple direct hits in a row, its armour penetration and damage will increase."
```

I dunno how does an "electronic autoloader" work on a weapon that you must use the chambering button to reload.

Reply:
I imagine it automatically takes the round out of the magazine and chambers it with only a button press

but I did not write this desc so 🤷